### ISSUE TITLE
space-rocks broken on Ubuntu 22.04.1

### ISSUE THREAD

**Issue by Siltala** [maxmahlke/rocks#10]
Hi,

I wanted to set up rocks on a server of mine running Ubuntu 22.04.1 (arm64 version, if it matters!) but I ran into unexpected trouble that I have not seen on my laptop using Arch Linux. Python in this case is 3.10.6 and space-rocks is 1.6.5.

I installed rocks with `pip install -U space-rocks` as normal, which completed without issue.
However, any usage of rocks fails with the following error:
```
$ rocks spins Ceres 
The asteroid name-number index is malformed. Run '$ rocks status' to update it.
```

Running the suggested `rocks status` fails with the following:
```
$ rocks status 
Traceback (most recent call last):
  File "/home/ubuntu/.local/bin/rocks", line 8, in <module>
    sys.exit(cli_rocks())
  File "/usr/lib/python3/dist-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/rocks/cli.py", line 161, in status
    date_index = index.get_modification_date()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/rocks/index.py", line 369, in get_modification_date
    date_index = (config.PATH_INDEX / "1.pkl").stat().st_mtime
  File "/usr/lib/python3.10/pathlib.py", line 1097, in stat
    return self._accessor.stat(self, follow_symlinks=follow_symlinks)
FileNotFoundError: [Errno 2] No such file or directory: '/home/ubuntu/.cache/rocks/index/1.pkl'
```

It looks like `rocks` is failing to initialize the cache for some reason (the ~/. cache/rocks folder does not exist at all?)
It's probably a trivial thing somewhere - can you give any suggestions for a fix?

Sorry if it's a dumb question and I'm missing something obvious!

**Comment by Siltala** [comment]
I confirm that manually creating an empty file in the path `/home/ubuntu/.cache/rocks/index/1.pkl` was enough to allow `rocks status` to work and to build the name-number index. Another bug came up in the process, which I was able to actually fix (and will submit a separate PR for)

**Comment by maxmahlke** [comment]
Hi Lauri, thanks for the issues and the PR. This issue and #13 are due to bugs I introduced in version 1.6.5, where I refactored the entire code base to make maintenance a bit easier and to speed up the command line interface by avoiding unnecessary imports.

For #13, the pandas imported in the PR are the correct way, great that you figured it out and could get it running on your side again.

Concerning this issue, rocks checks on every startup that the cache and index directory exist, and creates them if it's not the case. I simply forgot to reactivate this check during the refactoring. The fix you submitted in the PR works on the wrong level: instead of catching all the cases where the index might not exist with `try`-`except` clauses, I ensure the existence on startup. There is little you can do with rocks without the name-number index, so this makes more sense IMHO.

Now I appreciate the PR but because you mixed the two issues there and I cannot merge your solution to this issue, I will reject it. Please continue to submit PRs, and please make one PR per issue from now on :-)

This issue and #13 are fixed in version 1.6.6.I'm looking into #12 now and will push the version once I figured it out.
